tutorials/joint_controller.md

```diff
@@ -0,0 +1,615 @@
+\page jointcontrollers Joint Controllers
+
+Gazebo provides three joint controller plugins. Let's see a detailed description of each of them and example usage to help users to select right joint controller for their usage.
+# 1) JointController
+
+- Joint controller which can be attached to a model with a reference to a single joint.
+- Currently, only the first axis of a joint can be actuated.
+
+### Modes of JointController
+
+1) Velocity mode.
+
+2) Force mode.
+
+### Velocity mode:
+This mode lets the user control the desired joint velocity directly.
+
+### Force mode:
+
+A user who wants to control joint velocity using a PID controller can use this mode.
+
+This mode lets the user explicitly set the values of PID gains and also bounds for velocity.
+
+**Note**: This force mode is for the user who looking to manually tune PID gains for velocity control according to a specific use case (e.g. Custom models). For general testing purposes, velocity mode will give the best results.
+
+All the parameters related to this controller can be found [here](https://gazebosim.org/api/gazebo/7.0/classignition_1_1gazebo_1_1systems_1_1JointController.html#:~:text=joint%20is%20actuated.-,System%20Parameters,-%3Cjoint_name%3E%20The).

```

Comment:
TIL [Text Fragments](https://developer.mozilla.org/en-US/docs/Web/Text_fragments#syntax), neat!

However, to avoid having to change the url for every version of gz-sim, I recommend using the Doxygen keyword `\ref`. In this case, I believe it would be `\ref gz::sim::systems::JointController`.

btw, you can preview the documentation by going to `doxygen/html/index.html` in your build directory. See https://gazebosim.org/api/sim/7/install.html#autotoc_md94